### PR TITLE
Added unit-tests for CityService, CityAdminService

### PR DESCRIPTION
**Pull request by Lunaflumen**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.Tests/Services/City/CityServiceTests.cs [hunk 1]
```diff
@@ -355,11 +901,21 @@ private Region GetTestRegion()
...
+            return city;
+        }
+
+        private DataAccessCity.City GetTestCityWithoutLogo()
+        {
+            var city = new DataAccessCity.City
             {
-                Name = "city",
-                Logo = "710b8b06-6869-45db-894f-7a0b131e6c6b.jpg"
+                Name = "city"
```
Comment (tetyanakutsko) [inline EPlast/EPlast.Tests/Services/City/CityServiceTests.cs:915]: Unnecessary spaces

File: EPlast/EPlast.Tests/Services/City/CityServiceTests.cs [hunk 2]
```diff
@@ -212,6 +389,274 @@ public async Task EditAsync_ReturnsCityEdited()
...
+            CityService cityService = CreateCityService();
+
+            //// Act
+            var result = await cityService.GetCities();
+
+            //// Assert
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<IEnumerable<CityForAdministrationDTO>>(result);
+        }
+
```
Comment (tetyanakutsko) [inline EPlast/EPlast.Tests/Services/City/CityServiceTests.cs:657]: Unnecessary spaces

**Review by tetyanakutsko — APPROVED** [review]

File: EPlast/EPlast.Tests/Services/City/CityAdministrationServiceTests.cs [hunk 3]
```diff
@@ -44,22 +44,45 @@ public async Task GetAdministrationByIdAsync_ReturnsAdministrations()
...
             // Act
             var result = await _cityAdministrationService.GetAdministrationByIdAsync(It.IsAny<int>());
 
             // Assert
             Assert.NotNull(result);
+            foreach(CityAdministrationDTO admin in result)
+            {
+                Assert.NotNull(admin.User);
+                Assert.NotNull(admin.AdminType);
+            }
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/City/CityAdministrationServiceTests.cs:62]: Could you use LINQ ?

File: EPlast/EPlast.Tests/Services/City/CityServiceTests.cs [hunk 4]
```diff
@@ -142,6 +157,38 @@ public async Task GetCityProfileAsync_ReturnsCityProfile()
...
+            Assert.Null(result);
+        }
+
+        [Test]
+        public async Task GetCityProfileAsync_WhereAdminIsNull_ReturnCityProfile()
+        {
+            //// Arrange
+            CityService cityService = CreateCityService();
+            _mapper.Setup(m => m.Map<DataAccessCity.City, CityDTO>(It.IsAny<DataAccessCity.City>()))
+                .Returns(CreateFakeCityDtoWithoutAdmin(1).FirstOrDefault());
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/City/CityServiceTests.cs:181]: magic number

File: EPlast/EPlast.Tests/Services/City/CityServiceTests.cs [hunk 5]
```diff
@@ -184,6 +261,38 @@ public async Task GetCityAdminsAsync_ReturnsCityAdmins()
...
+            Assert.Null(result);
+        }
+
+        [Test]
+        public async Task GetCityAdminsAsync_WhereAdminIsNull_ReturnCity()
+        {
+            //// Arrange
+            CityService cityService = CreateCityService();
+            _mapper.Setup(m => m.Map<DataAccessCity.City, CityDTO>(It.IsAny<DataAccessCity.City>()))
+                .Returns(CreateFakeCityDtoWithoutAdmin(1).FirstOrDefault());
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/City/CityServiceTests.cs:285]: ?

File: EPlast/EPlast.Tests/Services/City/CityServiceTests.cs [hunk 6]
```diff
@@ -198,6 +307,74 @@ public async Task GetCityDocumentsAsync_ReturnsCityDocuments()
...
+            //// Assert
+            Assert.Null(result);
+        }
+
+        [Test]
+        public async Task GetLogoBase64_ReturnLogoBase64()
+        {
+            //// Arrange
+            CityService cityService = CreateCityService();
+            _cityBlobStorage.Setup(c => c.GetBlobBase64Async(It.IsAny<string>())).ReturnsAsync("LogoBase64");
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/City/CityServiceTests.cs:330]: same

**Review by voroniak — CHANGES_REQUESTED** [review]

**Review by bodnarira — APPROVED** [review]

**Review by voroniak — APPROVED** [review]

**Pull request merged**